### PR TITLE
DO NOT MERGE: Snippet to add ID to User

### LINKED ISSUE #449: Add option for Android device ID generation
See https://github.com/getsentry/sentry-java/pull/1455#issuecomment-831257971

### PR DESCRIPTION
**Pull request by ueman**
## :scroll: Description
See https://github.com/getsentry/sentry-dart/issues/449

https://sentry.io/share/issue/9aba382a0ae649baafaea54522729709/

## :bulb: Motivation and Context
Fixes https://github.com/getsentry/sentry-dart/issues/449

## :pencil: Checklist

- [ ] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] I updated the docs if needed
- [ ] All tests passing
- [ ] No breaking changes

### CODE REVIEW

File: flutter/example/lib/main.dart [hunk 1]
```diff
@@ -14,16 +15,30 @@ Future<void> main() async {
...
     appRunner: () => runApp(MyApp()),
   );
 }
 
+FutureOr<SentryEvent> addAndroidDeviceId(
+  SentryEvent event, {
+  dynamic hint,
+}) async {
+  var info = await DeviceInfoPlugin().androidInfo;
+  var user = (event.user ?? SentryUser()).copyWith(id: info.fingerprint);
```
Comment (ueman) [inline flutter/example/lib/main.dart]: There's a couple of candidates for Ids and I'm not quite sure which is should be user for which field.
I guess User.Id should be [fingerprint](https://pub.dev/documentation/device_info_plus_platform_interface/latest/model_android_device_info/AndroidDeviceInfo/fingerprint.html)
and Device.id should be [androidId](https://pub.dev/documentation/device_info_plus_platform_interface/latest/model_android_device_info/AndroidDeviceInfo/androidId.html).

Is [id](https://pub.dev/documentation/device_info_plus_platform_interface/latest/model_android_device_info/AndroidDeviceInfo/id.html) relevant? I guess not.

File: flutter/example/lib/main.dart [hunk 2]
```diff
@@ -14,16 +15,30 @@ Future<void> main() async {
   await SentryFlutter.init(
     (options) {
       options.dsn = _exampleDsn;
-      // use breadcrumb tracking of WidgetsBindingObserver
```
Comment (ueman) [inline flutter/example/lib/main.dart:17]: This is not needed anymore.

File: flutter/example/lib/main.dart [hunk 3]
```diff
@@ -14,16 +15,30 @@ Future<void> main() async {
...
 
+FutureOr<SentryEvent> addAndroidDeviceId(
+  SentryEvent event, {
+  dynamic hint,
+}) async {
+  var info = await DeviceInfoPlugin().androidInfo;
+  var user = (event.user ?? SentryUser()).copyWith(id: info.fingerprint);
+
+  // todo there's no ID field on SentryDevice
+  // var device = (event.contexts.device ?? SentryDevice()).copyWith();
```
Comment (marandaneto) [inline flutter/example/lib/main.dart]: that's ok, it's not gonna affect the number of unique users or anything, so its not a breaking change as it does not even exist in the Dart/Flutter SDK

Comment (ueman) [inline flutter/example/lib/main.dart]: Do we want to add the ID field for Device?

Comment (ueman) [reply flutter/example/lib/main.dart]: No, see https://github.com/getsentry/sentry-dart/pull/458#discussion_r629100195

File: flutter/example/lib/main.dart (hunk 1 above)
Comment (marandaneto) [reply flutter/example/lib/main.dart]: we only care for `user.id` which is using the value of the SDK 4.x
we must set only the value read from https://developer.android.com/reference/android/provider/Settings.Secure#ANDROID_ID

ignore `device.id`

File: flutter/example/lib/main.dart (hunk 2 above)
Comment (marandaneto) [reply flutter/example/lib/main.dart:17]: could be removed in a different PR as this is not going to be merged anyway

Comment (marandaneto) [reply flutter/example/lib/main.dart:17]: could be removed in a different PR, this PR is not going to be merged anyway

**Comment by marandaneto** [comment]
the other option instead of using event processor would be something like:

```
SentryUser user = new SentryUser();
user.setId(Settings.Secure.ANDROID_ID); // read using device_info

Sentry.setUser(user);
```

but its not a good idea because it'd affect iOS users, so Event processors would make more sense

**Pull request closed without merging**